# Problem Statement-1

# 1. A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions

# Importing Libraries:-

In [2]:
import pandas as pd
from scipy import stats
import numpy as np
from scipy.stats import chi2_contingency,chisquare

# Importing CSV File

In [4]:
ls

 Volume in drive C is ssd
 Volume Serial Number is 344C-651C

 Directory of C:\Users\phoneclicker\Downloads\excelr_kd\ASSIGNMENT\Hypothesis

01/06/2023  09:07 AM    <DIR>          .
01/06/2023  09:07 AM    <DIR>          ..
01/06/2023  09:06 AM    <DIR>          .ipynb_checkpoints
11/29/2022  03:33 PM                87 BuyerRatio.csv
11/29/2022  03:33 PM            13,423 Costomer+OrderForm.csv
11/29/2022  03:33 PM               525 Cutlets.csv
11/29/2022  03:33 PM            61,688 Hypothesis_Testing_Assignment.pptx
01/06/2023  09:07 AM            20,886 Hypothesis_Testing-3.ipynb
11/29/2022  03:33 PM             3,487 LabTAT.csv
               6 File(s)        100,096 bytes
               3 Dir(s)  23,076,634,624 bytes free


In [6]:
cutlet=pd.read_csv("Cutlets.csv")
cutlet.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

# EDA

In [8]:
cutlet.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [9]:
cutlet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [11]:
cutlet.duplicated().sum()

0

In [13]:
cutlet.value_counts()

Unit A  Unit B
6.4376  7.5093    1
7.2828  7.3070    1
7.0852  7.5459    1
7.1560  7.4220    1
7.1561  6.6965    1
7.2163  7.0133    1
7.2705  7.4314    1
7.2783  7.1180    1
7.2854  7.1688    1
6.9952  6.7594    1
7.3012  6.7878    1
7.3495  6.7478    1
7.3871  6.8110    1
7.3930  6.8810    1
7.3943  6.5780    1
7.4488  7.1522    1
7.0621  6.6606    1
6.9438  6.8889    1
6.5341  6.5217    1
6.8236  7.0503    1
6.5797  7.1581    1
6.5970  6.6672    1
6.6801  6.9182    1
6.6840  7.2402    1
6.7794  7.0992    1
6.8090  6.7703    1
6.8394  7.0240    1
6.9431  6.3346    1
6.8568  6.9399    1
6.8755  7.2212    1
6.9157  6.7300    1
6.9246  6.7652    1
6.9256  6.0380    1
6.9405  7.3875    1
7.5169  7.4059    1
dtype: int64

# Two sample T-test

In [16]:
X=cutlet["Unit A"]
Y=cutlet["Unit B"]

cutlets unit A -> X

cutlets unit B -> Y

H0 -> X==Y

H1 -> X!=Y


In [17]:
stats.ttest_ind( X,Y,alternative="two-sided")

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.47223947245995)

HO->47.22




H1->100-47.22

# Inferences

T Statistic = 0.72

sigma = 5

H0 -> 47.22%

H1 -> 52.78%

# Problem statement - 2

# A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.


# Importing CSV File

In [19]:
Tat=pd.read_csv('LabTAT.csv')

H0 -> x1=x2=x3=x4

h1 -> x1!=x2!=x3!=x4


In [20]:
Tat.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

# P-Value

In [22]:
p_value=stats.f_oneway(Tat.iloc[:,0],Tat.iloc[:,1],Tat.iloc[:,2],Tat.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [23]:
p_value[1]

2.1156708949992414e-57

In [24]:
if p_value[1]<0.05:
    print("H0 is rejected and there is difference of tat of all laboratories")
else:
    print("H0 is accepted and there is no difference of tat of all laboratories")

H0 is rejected and there is difference of tat of all laboratories


# Problem Statement - 3



# Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


# Importing CSV File

In [26]:
br=pd.read_csv('BuyerRatio.csv')

H0 -> Male = Female

h1 -> Male != Female


In [27]:
br

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [28]:
obs=np.array([[50,142,131,70],[435,1523,1356,750]])

In [29]:
chi2_contingency(obs)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [31]:
pValue=chi2_contingency(obs)[1]

In [32]:
pValue

0.6603094907091882

In [33]:
if pValue<0.05:
    print("H0 is rejected and there is no Male and female buyers are similar across in region")
else:
    print("H0 is accepted and Male and female buyers are similar across in region")

H0 is accepted and Male and female buyers are similar across in region


# Problem Statement - 4

# TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [42]:
co=pd.read_csv('Costomer+OrderForm.csv')
co

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [35]:
cof.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [43]:
print(co['Phillippines'].value_counts())
print(co['Indonesia'].value_counts())
print(co['Malta'].value_counts())
print(co['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64
Error Free    267
Defective      33
Name: Indonesia, dtype: int64
Error Free    269
Defective      31
Name: Malta, dtype: int64
Error Free    280
Defective      20
Name: India, dtype: int64


In [45]:
obs1_=np.array([[271,267,269,280],[29,33,31,20]])
obs1_

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [46]:
chi2_contingency(obs1_)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [47]:
pValue1=chi2_contingency(obs1_)[1]

In [40]:
pValue1

0.2771020991233135

In [48]:
if pValue<0.05:
    print("H0 is rejected")
else:
    print("H0 is accepted")
    

H0 is accepted
